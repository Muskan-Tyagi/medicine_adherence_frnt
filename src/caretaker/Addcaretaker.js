import { FlatList, RefreshControl, View } from "react-native";
import React, { useEffect } from "react";
import { Button, ListItem, SearchBar } from "react-native-elements";
import Dialog from "react-native-dialog";
import { useFocusEffect } from "@react-navigation/native";
import {API_URL} from '@env'
import AsyncStorage from "@react-native-async-storage/async-storage";


const Addcaretaker = ({navigation}) => {


    const [caretakers, caretakerstate] = React.useState([]);
    const [addcaretaker, addcaretakerstate] = React.useState(false);
    const [url, urlstate] = React.useState('');
    const [refresh , refeereshstate] = React.useState(false);

    const fetchcaretakers = async () =>  {
            const user_id = await AsyncStorage.getItem('user_id');
            fetch(`${API_URL}/api/caretaker/myCareTakers(Patient)?patient_id=${user_id}`)
                .then(resp => resp.json())
                .then(res => caretakerstate(res))
            
    }

    useEffect(()=>{
        
        fetchcaretakers()
    }
    ,[])

   const sendreqtouser = () => {

    

   }

    const renderitem = ({ item }) => {

        return (

            <ListItem style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' }}>
                <ListItem.Content>
                    <ListItem.Title>{item.caretaker_username}</ListItem.Title>
                    <ListItem.Subtitle>{item.created_at}</ListItem.Subtitle>
                </ListItem.Content>
                <Button title="Delete"></Button>
                
            </ListItem>
        )

    }

   
   

    return (
        <View style={{ flex: 1 ,backgroundColor:'white',height:'100%'}}>
           
            <FlatList data={caretakers} renderItem={renderitem} refreshControl={

<RefreshControl refreshing={refresh} onRefresh={fetchcaretakers}></RefreshControl>

            }></FlatList>
            <View style={{bottom:0,alignItems:'center'}}>
            <Button title="Add caretaker"
            buttonStyle={{backgroundColor:'#3743ab'}}
                onPress={() => navigation.getParent().navigate('Searchcaretaker') }

                style={{
                    backgroundColor: 'black',
                    borderWidth: 2,
                    position: 'absolute',
                    bottom: 10,
                    right: 10,
                    alignItems: 'center',
                    justifyContent: 'center',
                    elevation: 8,
                    borderColor: 'white',
                    borderRadius: 30,
                }}
                containerStyle={{
                    width: 200,
                    marginHorizontal: 50,
                    marginVertical: 10,
                }}
                titleStyle={{ fontWeight: 'bold' }}></Button>
                </View>
        </View>
    )


}

export default Addcaretaker;