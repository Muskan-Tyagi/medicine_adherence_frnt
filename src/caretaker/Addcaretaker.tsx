import { FlatList, RefreshControl, View,TouchableOpacity } from "react-native";
import React, { useEffect } from "react";
import { Avatar, Button, ListItem, SearchBar, SpeedDial } from "react-native-elements";
import Dialog from "react-native-dialog";
import { useFocusEffect } from "@react-navigation/native";
import {API_URL} from '@env';
import AsyncStorage from "@react-native-async-storage/async-storage";
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
import { faAngleRight } from '@fortawesome/free-solid-svg-icons'
import { Card} from 'react-native-paper';
import {IconProp} from '@fortawesome/fontawesome-svg-core';

interface Props{
    navigation:any
}

const Addcaretaker:React.FC<Props> = ({navigation}:Props) => {


    const [caretakers, caretakerstate] = React.useState([]);
    const [addcaretaker, addcaretakerstate] = React.useState(false);
    const [url, urlstate] = React.useState('');
    const [refresh , refeereshstate] = React.useState(false);
    const [open, setOpen] = React.useState(false);
    const fetchcaretakers = async () =>  {
            const user_id = await AsyncStorage.getItem('user_id');
            fetch(`${API_URL}/api/caretaker/myCareTakers(Patient)?patient_id=${user_id}`)
                .then(resp => resp.json())
                .then(res =>{
                    console.log(res)
                    caretakerstate(res)
                } )
            
    }

    useEffect(()=>{
        
        fetchcaretakers()
    }
    ,[])

   const sendreqtouser = () => {

    

   }

    const renderitem = ({ item }) => {
        console.log(item.patient_id,"b")

        return (

            <Card  onPress={() => {
                navigation.navigate('Patientprofile')}} 
            style={{ borderRadius:20,
            margin:6,
            borderColor:'lightgrey',
            elevation:3,
            shadowColor:'#3743ab'}}>
         <View style={{flexDirection:'row',padding:0}}>
         <ListItem style={{ flexDirection: 'row',
                             justifyContent: 'space-between',
                             alignItems: 'center' 
                             ,padding:5}}>
          <Avatar 
             size={64}
             rounded source={
             {uri:'https://lh3.googleusercontent.com/a-/AOh14Gg1r55ukyjleOVcBDEuTUt283ClmJE4ZSeFOSmD=s96-c'}}>

             </Avatar>
             <ListItem.Content>

                 <ListItem.Title style={{fontSize:16,marginLeft:3,fontWeight:'bold'}}
                 >{item.caretaker_username}
                 </ListItem.Title>
                 <ListItem.Subtitle>{' Created At :' + item.created_at}</ListItem.Subtitle>

             </ListItem.Content>

             <TouchableOpacity onPress={() => {}} 
             style={{paddingVertical: 15,}}>
                 <View style={{ alignItems: 'center'}}>
                     <FontAwesomeIcon icon={faAngleRight as IconProp} color={'black'} size={17} />

                 </View>
             </TouchableOpacity>
         </ListItem>
         </View>
     </Card>
   
        )

    }

   
   

    return (
        <View style={{ flex: 1 ,backgroundColor:'white',height:'100%'}}>
           
            <FlatList data={caretakers} renderItem={renderitem} refreshControl={

            <RefreshControl refreshing={refresh} onRefresh={fetchcaretakers}></RefreshControl>

            }></FlatList>
            <View style={{bottom:0,alignItems:'center'}}>
            <SpeedDial
                    isOpen={open}
                    style={{backgroundColor:'white'}}
                    overlayColor='white'
                    buttonStyle={{backgroundColor:'#3743ab'}}
                    icon={{ name: 'edit', color: 'white' }}

                    openIcon={{ name: 'close', color: 'white' }}
                    onOpen={() => setOpen(!open)}
                    onClose={() => setOpen(!open)}
  >
    <SpeedDial.Action
      icon={{ name: 'add', color: 'white' }}
      title="Add Caretaker"
      style={{height:50,}}
      buttonStyle={{backgroundColor:'#3743ab'}}

      onPress={() => navigation.navigate('Search Caretaker')}
    />
    <SpeedDial.Action
      icon={{ name: 'delete', color: 'white' }}
      title="Delete"
      buttonStyle={{backgroundColor:'#3743ab'}}

      style={{height:50}}

      onPress={() => console.log('Delete Something')}
    />
  </SpeedDial>
            <Button buttonStyle={{backgroundColor:'white'}} title="A"
            ></Button>
                </View>
        </View>
    )


}

export default Addcaretaker;