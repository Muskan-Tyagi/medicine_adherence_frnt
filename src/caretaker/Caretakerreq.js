import {View} from 'react-native';
import React from 'react';

const CaretakerReq = () => {
  return <View></View>;
};

export default CaretakerReq;
