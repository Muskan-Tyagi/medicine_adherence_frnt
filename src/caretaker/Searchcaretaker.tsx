
import React from "react";
import { FlatList, View } from "react-native";
import { Avatar, Button, ListItem, SearchBar } from "react-native-elements";
import {API_URL} from '@env';
import AsyncStorage from "@react-native-async-storage/async-storage";
import Toast from 'react-native-toast-message';


const Searchcaretaker = ({navigation}) => {
    
    const [searchword , searchstate] = React.useState('');
    const [data , datastate] = React.useState([]);
    const [load , loadstate] = React.useState(false);
    const [searchload , searchloadstate] = React.useState(false);

    const searchText = (text:any) => {
        searchstate(text)
        console.log(searchword)
         
    }


    const sendmailtouser = () => {
        searchloadstate(true)
        fetch(`${API_URL}/api/user/getbyemail?email=${searchword}&sender=Nikunj bisht`)
            .then(res => res.json())
            .then(resp=>{
                console.log(resp)
                searchloadstate(false)
                
                datastate([resp])
            })
            .catch(err =>{
                console.log(err);
                searchloadstate(false)
                Toast.show(
                    {
                        type:'success',
                        text1:'Invitation mail has been sent successfully!',
                        position:'bottom'
                    }
                )
            });



    }

    const sendreqtocaretaker = async (caret_id :String , caret_username : String) => {
      const pnt_id = await  AsyncStorage.getItem('user_id');
      const pt_name = await AsyncStorage.getItem('user_name');
       console.log(caret_id , pnt_id , pt_name);

       fetch(`${API_URL}/api/caretaker/savecaretaker`,{
           method:'POST',
           body:JSON.stringify({
            
                caretaker_id: caret_id,
                caretaker_username: caret_username,
                patient_id: pnt_id,
                patient_name: pt_name,
                req_status: false,
                sent_by: "P"
              
           }),
           headers:{
            "Content-type": "application/json"

           }
       })
       .then(res=>{
        console.log(res.json())
navigation.pop(1);
       })
       .catch((err)=>console.log(err))

    }

    const renderitem = ({ item }) => {

        return (

            <ListItem style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' }}>
                <ListItem.Content>
                <Avatar rounded source={{uri:'https://i.stack.imgur.com/l60Hf.png'} }></Avatar>
                    <ListItem.Title>{item.user_name}</ListItem.Title>
                    <ListItem.Subtitle>{item.email}</ListItem.Subtitle>
                </ListItem.Content>
                <Button loading={load} title="Send request" buttonStyle={{backgroundColor:'#3743ab'}}
                        onPress={()=> {

                            sendreqtocaretaker(item.user_id,item.user_name);
                            
                        }}></Button>                
            
                        
            </ListItem>
        )

    }

   return (

    <View style={{margin:10,backgroundColor:'white',height:'100%'}}>
    <Toast></Toast>

    <SearchBar style={{}} placeholder="Search Caretaker.." value={searchword} onChangeText={searchText} ></SearchBar>
    <Button loading={searchload} buttonStyle={{backgroundColor:'#3743ab'}} title="Search" onPress={()=>sendmailtouser()} containerStyle={{marginTop:10}}></Button>
    <FlatList data={data} renderItem={renderitem}></FlatList>
    </View>


   )


}
export default Searchcaretaker;