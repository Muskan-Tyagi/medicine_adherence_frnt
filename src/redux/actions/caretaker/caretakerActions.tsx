import Types from '../adherence/allTypes';
import { logger } from "react-native-logs";

const defaultConfig = {
  levels: {
    debug: 0,
    info: 1,
    warn: 2,
    error: 3,
  },
  transportOptions: {
    colors: {
      info: "blueBright",
      warn: "yellowBright",
      error: "redBright",
    },
  }
};
const log = logger.createLogger(defaultConfig);

export function fetchCaretakers(user_id: string) {
  return {
    type: Types.GET_CARETAKERS,
    payload: user_id,
  };
}
export function fetchCaretakerssuccess(data) {
  log.info(data,'success');
  return {
    type: Types.Success_CareTAKER_REQUEST,
    payload: data,
  };
}
export function fetchCaretakerserror(error) {
  log.error(error, 'ac');
  return {
    type: Types.Failed_CareTAKER_REQUEST,
    payload: error,
  };
}
export const caretakerActions = {
  fetchCaretakers,
  fetchCaretakerssuccess,
  fetchCaretakerserror
}
