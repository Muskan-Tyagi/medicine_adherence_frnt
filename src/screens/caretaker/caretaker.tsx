/* eslint-disable react/self-closing-comp */
import {Text} from 'react-native';
import React from 'react';
import Addmedicine from '../addMedicine';
import {createStackNavigator} from '@react-navigation/stack';
import UserMed from '../userMed';
import { NavigationContainer } from '@react-navigation/native';
const Stack = createStackNavigator();

const CareTaker = ({navigation}) => {
  return (
    <NavigationContainer>
    <Stack.Navigator initialRouteName="Addmedicine">
      <Stack.Screen
        name="Addmedicine"
        component={Addmedicine}
        options={{
          headerShown: false,
          // testID="musk",
          headerRight: () => {
            return (
              <Text
              testID='click'
              onPress={() => navigation.getParent().navigate('Events')}>
                Click
              </Text>
            );
          },
        }}></Stack.Screen>
      <Stack.Screen
        name="UserMed"
        component={UserMed}
        options={{headerShown: true}}></Stack.Screen>
    </Stack.Navigator>
    </NavigationContainer>
  );
};

export default CareTaker;
