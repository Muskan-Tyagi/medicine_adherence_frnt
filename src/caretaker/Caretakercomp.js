import { createBottomTabNavigator } from "@react-navigation/bottom-tabs"
import Addcaretaker from "./Addcaretaker";
import React from "react";
import CaretakerReq from "./Caretakerreq";

const Tab = createBottomTabNavigator();

export default function Caretakercomp(){

  return (
      <Tab.Navigator>
      
          <Tab.Screen name="Caretakers" component={Addcaretaker}></Tab.Screen>
          <Tab.Screen name="CaretakerRequest" component={CaretakerReq}></Tab.Screen>

      </Tab.Navigator>
  )


}