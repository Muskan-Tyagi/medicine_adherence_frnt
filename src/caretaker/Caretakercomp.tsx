import Addcaretaker from './Addcaretaker';
import React from 'react';
import CaretakerReq from './Caretakerreq';
import {GoogleSignin} from '@react-native-google-signin/google-signin';
import {useFocusEffect} from '@react-navigation/native';
import {Alert} from 'react-native';
import {Tab, TabView} from 'react-native-elements';
import {Caretaker_nurse , Userfriend} from './AllIcons';

export default function Caretakercomp({navigation}) {
  const [index, setIndex] = React.useState(0);

  const [login, loginstate] = React.useState(false);

  useFocusEffect(() => {
    async function checkforlog() {
      const islogged = await GoogleSignin.isSignedIn();
      if (!islogged) {

        Alert.alert("Sign in first to use this feature","Click ok to proceed",[
          {
            text:"Ok",
            onPress:()=>{navigation.navigate('Login')},
          },
          {
            text:"Cancel",
            onPress:()=>{navigation.navigate('Home')},
          }
        ]);
      }
      console.log(islogged);
      loginstate(islogged);
    }

    checkforlog();
  });

  return (
    <>
      <Tab
        value={index}
        onChange={e => setIndex(e)}
        indicatorStyle={{
          backgroundColor: 'white',
          height: 3,
        }}
        style={{backgroundColor: '#3743ab'}}
        variant="primary">
        <Tab.Item
          title="Caretakers"
          containerStyle={{backgroundColor: '#3743ab'}}
          titleStyle={{fontSize: 12}}
          icon={Caretaker_nurse()}
        />
        <Tab.Item
          title="Caretaker request"
          titleStyle={{fontSize: 12}}
          containerStyle={{backgroundColor: '#3743ab'}}
          icon={Userfriend()}
        />
      </Tab>

      <TabView value={index} onChange={setIndex} animationType="spring">
        <TabView.Item style={{backgroundColor: 'white', width: '100%'}}>
          <Addcaretaker navigation={navigation}></Addcaretaker>
        </TabView.Item>
        <TabView.Item style={{backgroundColor: 'white', width: '100%'}}>
          <CaretakerReq></CaretakerReq>
        </TabView.Item>
      </TabView>
    </>
  );
}
