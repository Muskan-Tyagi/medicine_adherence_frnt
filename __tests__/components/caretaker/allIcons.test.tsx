import Enzyme, { render } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { Provider } from "react-redux";
import React from "react";
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import enableHooks from "jest-react-hooks-shallow";
import toJson from "enzyme-to-json";
import {Righttoobracket} from '../../../src/components/caretaker/allIcons'
enableHooks(jest);

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');


describe('Click send image', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<Righttoobracket/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
