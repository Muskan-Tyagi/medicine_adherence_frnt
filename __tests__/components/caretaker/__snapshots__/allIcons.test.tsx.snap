// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "white",
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  Û∞çÇ
</Text>
`;
